src/activities/menu/Menu.qml

```diff
@@ -218,12 +236,24 @@ ActivityBase {
                     }
 
                     function selectCurrentItem() {
-                        particles.burst(10)
-                        ActivityInfoTree.filterByTag(modelData.tag)
-                        ActivityInfoTree.filterLockedActivities()
-                        ActivityInfoTree.filterEnabledActivities()
-                        menuActivity.currentTag = modelData.tag
-                        section.currentIndex = index
+
+
+                               if(modelData.tag === "search"){
+
+                                   section.currentIndex = index
+                                   menuActivity.currentTag = modelData.tag
+                                   ActivityInfoTree.display(input.text);
```

Comment:
I would rename display to filterBySearch or something like that. display is too generic